src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CollectionInitializer.cs

```diff
@@ -3,64 +3,94 @@
             EnsureArg.IsNotNull(upgradeManager, nameof(upgradeManager));
+            EnsureArg.IsNotNull(retryExceptionPolicyFactory, nameof(retryExceptionPolicyFactory));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            _collectionId = collectionId;
+            _cosmosCollectionConfiguration = cosmosCollectionConfiguration;
             _cosmosDataStoreConfiguration = cosmosDataStoreConfiguration;
-            _initialCollectionThroughput = initialCollectionThroughput;
             _upgradeManager = upgradeManager;
+            _retryExceptionPolicyFactory = retryExceptionPolicyFactory;
+            _clientTestProvider = clientTestProvider;
             _logger = logger;
         }
 
         public async Task<Container> InitializeCollection(CosmosClient client)
         {
             Database database = client.GetDatabase(_cosmosDataStoreConfiguration.DatabaseId);
-            Container containerClient = database.GetContainer(_collectionId);
+            Container containerClient = database.GetContainer(_cosmosCollectionConfiguration.CollectionId);
 
-            _logger.LogInformation("Finding Container: {collectionId}", _collectionId);
-            var existingContainer = await database.TryGetContainerAsync(_collectionId);
+            _logger.LogInformation("Finding Container: {collectionId}", _cosmosCollectionConfiguration.CollectionId);
 
-            if (existingContainer == null)
-            {
-                _logger.LogInformation("Creating Cosmos Container if not exits: {collectionId}", _collectionId);
+            AsyncPolicy retryPolicy = _retryExceptionPolicyFactory.GetRetryPolicy();
+
+            var existingContainer = await retryPolicy.ExecuteAsync(async () => await database.TryGetContainerAsync(_cosmosCollectionConfiguration.CollectionId));
+
+            _logger.LogInformation("Creating Cosmos Container if not exits: {collectionId}", _cosmosCollectionConfiguration.CollectionId);
 
-                var containerResponse = await database.CreateContainerIfNotExistsAsync(
-                    _collectionId,
+            var containerResponse = await retryPolicy.ExecuteAsync(async () =>
+                await database.CreateContainerIfNotExistsAsync(
+                    _cosmosCollectionConfiguration.CollectionId,
                     $"/{KnownDocumentProperties.PartitionKey}",
-                    _initialCollectionThroughput);
+                    _cosmosCollectionConfiguration.InitialCollectionThroughput));
 
-                containerResponse.Resource.DefaultTimeToLive = -1;
+            containerResponse.Resource.DefaultTimeToLive = -1;
 
-                existingContainer = await containerClient.ReplaceContainerAsync(containerResponse);
+            existingContainer = await retryPolicy.ExecuteAsync(async () => await containerClient.ReplaceContainerAsync(containerResponse));
+
+            if (_cosmosCollectionConfiguration.InitialCollectionThroughput.HasValue)
```

Comment:
@johnstairs, sorry I missed this. We can only do this if the collection is new, else we risk overriding the existing settings. This is currently crashing CI as they are set to Autoscale